llvm/lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -1356,6 +1356,12 @@ void ExecutionEngine::emitGlobalVariable(const GlobalVariable *GV) {
   // Don't initialize if it's thread local, let the client do it.
   if (!GV->isThreadLocal()) {
     if (!GV->hasInitializer()) {
+      if (!isForPreExecute) {
```

Comment:
2 things:

1) There is no point in having 2 identical flags in both the base and derived class. The flag from the derived class should be moved in the base one.
2) I am not sure this whole thing is sound. What is the purpose of this?

Reply:
While PreExecuting, everything has to be intialized, so this commit:
https://github.com/leaningtech/cheerp-compiler/commit/13371bd3b865739391e61e27484935b85b1c88ef
adds the checks for debugging purposes.
Now when NON preexecuting, I think it's fine to just return, or even go to the end of the function.